backend/src/bundles/auth/auth.service.ts

```diff
@@ -11,10 +12,21 @@ class AuthService {
         this.userService = userService;
     }
 
-    public signUp(
+    private async signIn(
+        id: number,
+    ): Promise<{ id: number; email: string; token: string }> {
+        const user = await this.userService.find(id);
+        return {
+            ...user,
+            token: await token.create({ id }),
+        };
+    }
+
+    public async signUp(
         userRequestDto: UserSignUpRequestDto,
     ): Promise<UserSignUpResponseDto> {
-        return this.userService.create(userRequestDto);
+        const user = await this.userService.create(userRequestDto);
+        return this.signIn(user.id);
```

Comment:
```suggestion
        return await this.signIn(user.id);
```
You may need to add await to signIn, i think. 
Because this function is async signIn(...){}

Reply:
In this case you don't need to do it 
I have good article about it http://hassansin.github.io/Why-return-await-Is-a-Bad-Idea

Reply:
Hello, I think there is no need to await this.signIn(...) since I'm immediately returning it.